cmd/admin-trace.go

```diff
@@ -378,11 +458,161 @@ func matchingOpts(ctx *cli.Context) (opts matchOpts) {
 		fatalIf(probe.NewError(e).Trace(ctx.String("filter-size")), "Unable to parse input bytes.")
 	}
+
+	if ctx.String("rotate-trace-logs-size") != "" {
+		rotateTraceLogsSize, e = humanize.ParseBytes(ctx.String("rotate-trace-logs-size"))
+		fatalIf(probe.NewError(e).Trace(ctx.String("rotate-trace-logs-size")), "Unable to parse input trace logs bytes.")
+	}
+
+	if ctx.String("rotate-trace-logs-frequency") != "" {
+		rotateTraceLogsFrequency, e = strconv.ParseUint(ctx.String("rotate-trace-logs-frequency"), 0, 64)
+		fatalIf(probe.NewError(e).Trace(ctx.String("rotate-trace-logs-frequency")), "Unable to parse input trace logs frequency.")
+	}
+
+	if ctx.String("rotate-trace-logs-pattern") != "" {
+		rotateTraceLogsPattern = ctx.String("rotate-trace-logs-pattern")
+		traceLogsPattern = rotateTraceLogsPattern
+	}
+
+	if ctx.String("rotate-trace-logs-compress") != "" {
+		rotateTraceLogsCompress = ctx.String("rotate-trace-logs-compress")
+		traceLogsCompress = rotateTraceLogsCompress
+	}
+
 	opts.requestSize = requestSize
 	opts.responseSize = responseSize
+	opts.rotateTraceLogsSize = rotateTraceLogsSize
+	opts.rotateTraceLogsFrequency = rotateTraceLogsFrequency
+	opts.rotateTraceLogsPattern = rotateTraceLogsPattern
+	opts.rotateTraceLogsCompress = rotateTraceLogsCompress
 	return
 }
 
+// Write trace log file
+func writeTracelogsFile(log string) {
+	traceLogsFileSize += len(log)
+	if _, err := traceLogsFile.WriteString(log + "\n"); err != nil {
+		traceLogsFile.Close()
+		fatalIf(probe.NewError(err), "Unable to write trace log file `"+defineTraceLogsFile()+"`")
+	}
+}
+
+// Rotate trace log file
+func rotateTraceLogsFile() {
+	defer func() {
+		// Close current file
+		if err := traceLogsFile.Close(); err != nil {
+			fatalIf(probe.NewError(err), "Unable to close current trace log file `"+traceLogsFilePath+"`")
+		}
+	}()
+	// Compress current file
+	if traceLogsCompress != "" {
+		// Remove raw file
+		defer func() {
+			if err := os.Remove(traceLogsFilePath); err != nil {
+				fatalIf(probe.NewError(err), "Unable to remove current trace log file `"+traceLogsFilePath+"`")
+			}
+		}()
+		switch traceLogsCompress {
+		case "gzip":
+			compressGzip()
+		case "zstd":
+			compressZstd()
+		default:
+			break
+		}
+	}
+}
+
+// gzip
+func compressGzip() {
+	compressFile, err := os.OpenFile(traceLogsFilePath+".gz", os.O_CREATE|os.O_RDWR, 0o644)
+	if err != nil {
+		compressFile.Close()
+		fatalIf(probe.NewError(err), "Unable to open trace log file archive for gzip `"+traceLogsFilePath+"`")
+	}
+	traceLogsFile.Seek(0, 0)
+	writer := gzip.NewWriter(compressFile)
+	_, err = io.Copy(writer, traceLogsFile)
+	if err != nil {
+		writer.Close()
+		compressFile.Close()
+		fatalIf(probe.NewError(err), "Unable to write trace log file archive for gzip `"+traceLogsFilePath+"`")
+	}
+	writer.Close()
+	compressFile.Close()
+}
+
+// zstandard
+func compressZstd() {
+	compressFile, err := os.OpenFile(traceLogsFilePath+".zst", os.O_CREATE|os.O_RDWR, 0o644)
```

Comment:
Maybe needed if we append:
```suggestion
	compressFile, err := os.OpenFile(traceLogsFilePath+".zst", os.O_CREATE|os.O_RDWR|os.O_APPEND, 0o644)
```